inspire/base/templates/header.html

```diff
@@ -128,14 +104,37 @@
   </nav>
 </section>
 
-{% if request.path  == "/search" %}
+{% if request.path  == "/search" or request.path | is_collection_path %}
+
+  {% if request.path  == "/search" %}
+    {% set collection_name = request.args.get('cc') %}
+  {% else %}
+    {% set collection_name = request.path | check_collection %}
```

Comment:
I don't think you need a filter here, the request object should have enough information (http://flask.pocoo.org/docs/0.10/api/#flask.Request.url_root)